pkg/controller/atlasdatabaseuser/atlasdatabaseuser_controller.go

```diff
@@ -127,7 +127,17 @@ func (r *AtlasDatabaseUserReconciler) Reconcile(ctx context.Context, req ctrl.Re
 		return result.ReconcileResult(), nil
 	}
 
-	dus, err := dbuser.NewAtlasDatabaseUsersService(ctx, r.AtlasProvider, project.ConnectionSecretObjectKey(), log)
+	hasFallback := r.AtlasProvider.HasGlobalFallbackSecret()
+	credentialsSecret, err := customresource.ComputeSecretWithFallback(hasFallback, project, databaseUser)
```

Comment:
Does this not change the default behaviour going forward? Are we not now attempting to fall back to the global secret at all times?

Reply:
This would allow not having a global, whereas now the global is always expected. But the default behaviour should remain that there is the project creds or the global ones.